src/lib/utils/migration.ts

```diff
@@ -0,0 +1,75 @@
+import { ServiceFactory, LocalStorageService } from "../classes";
+import { ChainData } from "../classes/services/chainsService";
+import { MAX_CACHED_BLOCKS } from "../constants";
+import { LocalStorageKey } from "../enums";
+
+let storageService: LocalStorageService;
+
+type Migration = () => void;
+const MIGRATIONS: Record<number, Migration> = {
+    1: migrateToVersion2,
+};
+
+const CURRENT_PERSISTED_DATA_VERSION = 2;
+
+/**
+ *
+ */
+function initStorageService(): void {
+    storageService = ServiceFactory.get<LocalStorageService>(LocalStorageService.ServiceName);
+}
+
+/**
+ *
+ */
+export function checkAndMigrate(): void {
+    if (!storageService) {
+        initStorageService();
+    }
+    const version = storageService.load(LocalStorageKey.PersistedDataVersion) ?? 1; // Get the current version
+    const shouldMigratePersistedChains = version < CURRENT_PERSISTED_DATA_VERSION;
+    if (shouldMigratePersistedChains) {
+        migrateEachVersion();
+    }
+}
+/**
+ *
+ */
+function migrateEachVersion(): void {
+    if (!storageService) {
+        initStorageService();
+    }
+    let version: number = storageService.load(LocalStorageKey.PersistedDataVersion) ?? 1; // Get the current version
+    for (CURRENT_PERSISTED_DATA_VERSION; version < CURRENT_PERSISTED_DATA_VERSION; version++) {
+        try {
+            MIGRATIONS[version]();
+            storageService.save(LocalStorageKey.PersistedDataVersion, version + 1);
+        } catch {
+            storageService.remove(LocalStorageKey.ShowHexAsText);
+            storageService.remove(LocalStorageKey.Chains);
```

Comment:
I dont think it makes sense to have this code here, that actually runs on every migration. This should be added to  `migrateToVersion2` as a catch of each relevant section. Also, `showHexAsText` potentially will not be removed so it needs to be taken into account in the catch 🙏🏼